src/OpenTelemetry/OpenTelemetry.csproj

```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <!-- OmniSharp/VS Code requires TargetFrameworks to be in descending order for IntelliSense and analysis. -->
-    <TargetFrameworks>netstandard2.0;net462</TargetFrameworks>
+    <TargetFrameworks>net6.0;netstandard2.1;netstandard2.0;net462</TargetFrameworks>
```

Comment:
New targets! 🙈 

I know no one wants to do this, but it is a necessary evil to keep current 😁 

Example of runtime doing the same thing: https://github.com/dotnet/runtime/blob/1a02f0d62481b9830ded42d514053aab8b3aca3c/src/libraries/System.Diagnostics.DiagnosticSource/src/System.Diagnostics.DiagnosticSource.csproj#L101

Why these particular targets?

* The nullable attributes are available in runtime from netstandard2.1+.
* `IsExternalInit` is available net5.0+.

What this project needs to do is not include the shims when they are natively part of runtime. Otherwise projects consuming SDK internals & trying to do something like `[AllowNull]` will get a warning about the type existing in runtime & SDK.